great_expectations/rule_based_profiler/attributed_resolved_metrics.py

```diff
@@ -45,6 +45,16 @@ def _detect_illegal_array_type_or_shape(values: MetricValues) -> bool:
                     ):
                         return True
 
+                # Components of different types cannot be packaged into "numpy.ndarray" type (due to type mismatch).
+                values_iterator: Iterator = iter(values)
+                first_value_type: type = type(next(values_iterator))
+                current_type: type
+                if not all(
+                    type(current_value) == first_value_type
+                    for current_value in values_iterator
+                ):
+                    return True
```

Comment:
Why `type` over `isinstance`? Also, instead of doing the conditional and returning `True`, can we just return the conditional?

Reply:
@cdkini I suspect that using `isinstance` will not work; the goal here is to insure that all elements of the iterable have exactly the same type.  We return boolean `True` as soon as any necessary condition for failure is satisfied (so `True` here means "fail").  Thanks.

Reply:
We don't care about inheritance at all here right? `type` will give us a string but `isinstance` will respect polymorphism.

This seems fine to me.